api_app/db/repositories/resources.py

```diff
@@ -177,7 +177,7 @@ def validate_patch(self, resource_patch: ResourcePatch, resource_template_repo:
         update_template["required"] = []
         update_template["properties"] = {}
         for prop_name, prop in enriched_template["properties"].items():
-            if ("updateable" in prop.keys() and prop["updateable"] is True):
+            if (resource_action == "install" or ("updateable" in prop.keys() and prop["updateable"] is True)):
```

Comment:
Passing comment: It's annoying that this talks about "updateable" when, I assume, this is checking for upgrades?

TBH...what is this code saying?
Your change is checking "the action is install", but what was the existing code doing, and is it a good idea to combine your code with this code?
Would it be cleaner (e.g. if other resource actions are supported later) to have a `elif` even if the code block associated is the same?

Reply:
The existing code only allows properties marked as updateable to be updated in a pipeline. This PR adds the ability for them to be set if the action is install - be they marked updatable or not. Not sure an `elif` helps much, as it just repeats the next line that does the assignment.

Reply:
It helps make it clear there are two possibile situations:

1. The resource action is "install"
2 Marker as updateable

I agree...logically this one-liner does the same thing, in fewer lines of code. I'm just saying that comparing two very different things (both which use different concepts) wasn't super clear to me. 

Also, I thought there were plans to extend this to other resource actions in the future. Is that not the case? If it is, then it makes it more obvious where we'd add those additional clauses.

It's not a big thing, but combination of an `if`, an `or`, and an `and` within the brackets, along with testing two different things, made this one-liner look messy.

Reply:
Happy to change if think its clearer.

Reply:
For other resource actions, it needs to adhere to it being updatable or not. The only exception would be the first time a resource is deployed - hence `install`. I might be missing something, but don't think so.

Reply:
Also, is the second clause basically doing...

```python
prop.get("updateable", False)
```

...?

Reply:
Look that way :)

Reply:
1. Aren't there cases where a template has a default we don't want to allow changing in an install?
2. I agree the condition is somewhat confusing. I remember a discussion once to allow setting a property just once, maybe this is the case for that (too)?

Reply:
```suggestion
            if (resource_action == "install" or prop.get("updateable", False) is True):
```

I assume that the `is True` is there because we're not interested in it being "truthy". Instead, we want it to be a boolean type, and specifically `True`. Otherwise, we could drop the `is True`.

Reply:
My suggestion maintains the specific semantics around `is True`. 
Yours would start to allow truthy values like "1" to be seen as `True`.

Reply:
This make you happier?

```suggestion
           if resource_action == "install" or prop.get("updateable", False) is True:
```

Reply:
Done